### ISSUE TITLE
Some tests fails, because they're unable to find the topology.env file. + fix

### ISSUE THREAD

**Issue by hatesmandatoryregistration** [open-mpi/hwloc#369]
### What version of hwloc are you using?

The newest just downloaded from this git and built.

/usr/local/bin/lstopo --version
lstopo 2.2.0a1-git

git show
commit 3840d8c5e11b3f1317f4264216c14dbb8ab2c920

### Which operating system and hardware are you running on?

RHEL-8.2, Linux 4.18.0

### Details of the problem

Some tests are failing unexpectedly.

[hostname x86+linux]# ./test-topology.sh 64amd64-4p2n4ca2co.topoextnuma.output
./test-topology.sh: line 115: source: 64amd64-4p2n4ca2co.topoextnuma.env: file not found
[hostname x86+linux]# echo $?
1

### Additional information

The line:
        source "$topology".env
looks right and it is under bash. But these test scripts are built with /bin/sh [1] which doesn't try to find file to source in ./ (it uses just $PATH). Everything works properly after applying this simple patch to all 3 instances of test-topology.sh.in:

``` diff
diff -r hwloc_orig/hwloc/tests/hwloc/linux/test-topology.sh.in hwloc_fixed/hwloc/tests/hwloc/linux/test-topology.sh.in
122c122
<     source "$topology".env
---
>       source "$(pwd)/$topology".env
diff -r hwloc_orig/hwloc/tests/hwloc/x86/test-topology.sh.in hwloc_fixed/hwloc/tests/hwloc/x86/test-topology.sh.in
109c109
<     source "$topology".env
---
>       source "$(pwd)/$topology".env
diff -r hwloc_orig/hwloc/tests/hwloc/x86+linux/test-topology.sh.in hwloc_fixed/hwloc/tests/hwloc/x86+linux/test-topology.sh.in
115c115
<     source "$topology".env
---
>       source "$(pwd)/$topology".env
```

I'm able to push the fix to this git, but I'd like to ask if it's a good idea first.
Is it the way how you want to fix it? Or do you prefere to switch /bin/sh to /bin/bash or something else?

Thank you.

[1] Build process:
git clone https://github.com/open-mpi/hwloc/
./autogen.sh
./configure
make

**Comment by bgoglin** [comment]
Hello
@BASH@ gets replaced with /bin/bash as expected here.
    AC_PATH_PROG([BASH], [bash]) # in config/hwloc.m4
Any idea why it's replaced with /bin/sh on your machine?
I don't have a RHEL8 yet. All our RHEL6/7, Fedora up to 29, Debian up to 9, Ubuntu up to 18.04 don't have this problem.
Thanks
Brice

**Comment by hatesmandatoryregistration** [comment]
Hello Brice,
I have tested to build the hwloc on several systems.

It has created test-topology.sh with /bin/sh in RHEL 6.10, RHEL 7.[2345678] and RHEL 8.2.
The right /bin/bash was there in Debian 9.3 and in RHEL7 after I have changed the symlink /bin/sh to /bin/dash (the same as Debian has) or /bin/zsh.

I have minimal experience in autoconf, automake and m4. I know they exists and how to run them to build something, but I have never wrote any code for them. 

It seems, that the behavior depends on what's in /bin/sh strongly. Other people have found similar problem here: http://archive.linuxfromscratch.org/mail-archives/lfs-dev/2007-March/059035.html

If you want to use bash, I suggest to fix the m4 or configure somehow (I have no idea how, but I can read through the mail archives I have linked to try to find a solution).

If you want to use /bin/sh, I suggest to use my patch which doesn't change behavior of the script while running in bash, but fixes it while running in /bin/sh.

Let me know how can I help with this.

Have a nice day

**Comment by bgoglin** [comment]
Indeed, changing my /bin/sh to bash makes our script use /bin/sh. I don't understand how autoconf ends up doing this. However this case should work fine anyway, since we still use bash in the end.
Your original issue on RHEL8 got /bin/sh but what did /bin/sh point to?

**Comment by hatesmandatoryregistration** [comment]
Hello Brice,
I have no idea how the autoconf does that actually. It chooses shell somehow and test its capabilities probably. My experiments revealed, that:
- /bin/sh is link to /bin/bash -> autoconf uses /bin/sh
- /bin/sh is link to /bin/dash, but /bin/dash is copy of /bin/bash -> autoconf uses /bin/sh
- /bin/sh is link to real /bin/dash (or /bin/zsh) -> autoconf uses /bin/bash

It looks like no problem at the first glance. It's bash anyway. But bash checks how it was called and when it was called as /bin/sh it emulates the ancient sh. Therefore source in the test-topology.sh scripts stops working, because bash emulating sh doesn't look to ./ (bash is looking there). That's why I have proposed to add $(pwd) in my patch. It wouldn't affect behavior under bash and fix the problem under sh.

Fixing the autoconf would be the best/right way of course.

Have a nice day.

**Comment by bgoglin** [comment]
OK. I can't reproduce the issue here anyway. The only way to get a failure is a have sh point to bash during configure (so that scripts use sh) and then change sh to dash before running scripts. However, when I do this, "source" and "$()" don't work anymore because they are bashims. So I wonder is the problem is not about sh, but rather about bash "posix" mode which would be enabled somehow on your machine.
Anyway, I am reworking your fix so it works when building outside of the source tree.
If you give me a proper name, I will be able to thank you in the commit message.

**Comment by hatesmandatoryregistration** [comment]
Hello Brice,
thank you for working on this issue.

I have played with bash, sh emulated by bash, zsh and dash a bit. It seems, that your bash doesn't emulate sh so much as RHEL's one. You're right, that 'source' is supported just by bash and zsh (from my list). But the $() works on all of them.

I suggest to change:
``` diff
source "$topology".env
```
in my patch to:
``` diff
. "$topology".env
```
which works on all shells I have tested.

Proper fix, which would force automake to use bash would be nice of course. But I have no idea how to do it and I'm not planning to learn autoconf soon.

My civil name is Erik Hamera.

Have a nice day
